File: CodeWars/.js/8kyu-Jennys-Secret-Message.js
// Jenny has written a function that returns a greeting for a user. However, she's in love with Johnny, and would like to greet him slightly different. She added a special case to her function, but she made a mistake.

// Can you help her?

describe("Jenny's greeting function", function(){
    it("should greet some people normally",function(){
      Test.assertEquals(greet("Jim"), "Hello, Jim!");
      Test.assertEquals(greet("Jane"), "Hello, Jane!");
      Test.assertEquals(greet("Simon"), "Hello, Simon!");
    });
    it("should greet Johnny a little bit more special",function(){
      Test.assertEquals(greet("Johnny"), "Hello, my love!");
    });
  });

//  my code

// 
// @param: name
// @return: statement, varying if the input name is Johnny
// 

function greet(name){
    return (name != 'Johnny') ? `Hello, ${name}!` : 'Hello, my love!'
}

File: CodeWars/.cs/8kyu-JennysSecretMessage.cs
// Jenny has written a function that returns a greeting for a user. However, she's in love with Johnny, and would like to greet him slightly different. She added a special case to her function, but she made a mistake.

// Can you help her?

using System;
using NUnit.Framework;

[TestFixture]
public class JennysGreeting
{
  [Test]
  public static void ShouldGreetJimNormally()
  {
    Assert.AreEqual("Hello, Jim!", Kata.greet("Jim"));
  }
  [Test]
  public static void ShouldGreetJaneNormally()
  {
    Assert.AreEqual("Hello, Jane!", Kata.greet("Jane"));
  }
  [Test]
  public static void ShouldGreetSimonNormally()
  {
    Assert.AreEqual("Hello, Simon!", Kata.greet("Simon"));
  }
  
  [Test]
  public static void ShouldGreetJohnnySpecially()
  {
    Assert.AreEqual("Hello, my love!", Kata.greet("Johnny"));
  }
}

// my code

public static class Kata
{
  public static string greet(string name) => (name == "Johnny") ? $"Hello, my love!" : $"Hello, {name}!";
}